chainerx_cc/chainerx/routines/loss.cc

```diff
@@ -24,4 +28,19 @@ Array HuberLoss(const Array& x1, const Array& x2, Scalar delta) {
     return Where(abs_a < delta_array, 0.5 * Square(a), delta * (abs_a - Scalar{0.5} * delta));
 }
 
+Array Hinge(const Array& x1, const Array& x2, const std::string& norm) {
+    int64_t num = x1.shape()[0];
+    Array bottom_diff = Copy(x1);
+    std::vector<Array> indices = {Arange(num), x2};
+    Array concat = Transpose(Concatenate(indices, 0));
+    IMultiply(At(bottom_diff, concat), -1);
```

Comment:
You can turn this into.
```CXX
Array selected = bottom_diff.At(concat);
selected *= -1;
```
As they are methods for `Array`. This should compile.

Reply:
Thanks @kshitij12345 .The only problem seems to be that `At` expects a `vector` with `ArrayIndex` as element type.

Reply:
I guess, you can create `ArrayIndex` using `Slice` object with your range of indices.

Reply:
`IMultiply` cannot be used for an array required grad.

Reply:
I think there is no need to use `At`.
How about implementing like as follows for example?
```py
Array one_minus_diff = Where(ExpandDims(x2, 1) == Arange(num, x2.dtype), 0, 2);
```